test/target_web/controllers/api/v1/topic_controller_test.exs

```diff
@@ -0,0 +1,114 @@
+defmodule TargetWeb.API.V1.TopicControllerTest do
+  use TargetWeb.ConnCase
+
+  alias Target.Targets
+  alias Target.Targets.Topic
+  alias TargetWeb.APIAuthPlug
+
+  @pow_config [otp_app: :target]
+  @create_attrs %{
+    name: "some name"
+  }
+  @update_attrs %{
+    name: "some updated name"
+  }
+  @invalid_attrs %{name: nil}
+
+  def fixture(:topic) do
+    {:ok, topic} = Targets.create_topic(@create_attrs)
+    topic
+  end
+
+  setup %{conn: conn} do
+    %{conn: conn, user_token: user_token} = create_user(conn)
+
+    conn =
+      conn
+      |> put_req_header("authorization", user_token)
```

Comment:
This would be cool to extract into a module when you have 2 or more controller tests

Reply:
you can do something like https://github.com/rootstrap/open-rbac/blob/develop/test/support/conn_case.ex using tags to mark tests as authenticated and just make the header set generic (not sure how to do it with pow though).